hail/src/main/scala/is/hail/expr/ir/agg/LinearRegressionAggregator.scala

```diff
@@ -13,7 +13,8 @@ object LinearRegressionAggregator {
   val vector = PCanonicalArray(scalar, true)
   val stateType: PCanonicalTuple = PCanonicalTuple(true, vector, vector, PInt32(true))
 
-  def resultType: PCanonicalStruct = PCanonicalStruct(required = true, "xty" -> vector, "beta" -> vector, "diag_inv" -> vector, "beta0" -> vector)
+  private val optVector = vector.setRequired(false)
```

Comment:
Why do we need `optVector` vs just making the original `vector` on line 13 not required?

Reply:
We want the state to have required vectors, I think.